route/option.go

```diff
@@ -328,6 +328,28 @@ func Sorter(
 	return sorterFunc(sorter)
 }
 
+// PlanMarshaller defines an interface that is used to override router's default
+// output when using multiple vehicles.
+type PlanMarshaller interface {
+	Marshal(PartialPlan) ([]byte, error)
+}
+
+// VehicleMarshaller defines an interface that is used to override router's
+// default output when using exactly one vehicle.
+type VehicleMarshaller interface {
+	Marshal(PartialVehicle) ([]byte, error)
```

Comment:
I would suggest we call the function `MarshalJSON` similar to the `Marshaler` interface in `encoding/json`.

Reply:
was removed in favor of other implementation